be/src/formats/parquet/column_converter.cpp

```diff
@@ -682,9 +682,16 @@ Status Int64ToDateTimeConverter::_convert_to_timestamp_column(const ColumnPtr& s
     for (size_t i = 0; i < size; i++) {
         dst_null_data[i] = src_null_data[i];
         if (!src_null_data[i]) {
-            Timestamp timestamp =
-                    timestamp::of_epoch_second(static_cast<int>(src_data[i] / _second_mask),
-                                               static_cast<int>((src_data[i] % _second_mask) * _scale_to_nano_factor));
+            int64_t seconds = src_data[i] / _second_mask;
+            // for the case if seconds < 0(anytime before 1970-01-01 00:00:00)
+            // like '1900-01-01 00:00:00', we will get '1899-12-31 23:54:17'
+            // there is 343 seconds behind somehow.(But I really don't know why)
+            // so to get correct answer, we have to add that back.
+            if (seconds < 0) {
```

Comment:
If the second is -1, the result will be larger than utc? Is that correct?

Reply:
if second is -1, you expect to get '1969-12-31 23:59:59', However you probably will get `1969-12-31 23:54:16`(343 seconds are lost). So to offset that, I'll add 343 seconds.

Reply:
emmm.... I think your method `of_epoch_second` is error...the method compute julian date is error
your refs this code:
```
    TimestampValue ep;
    ep.from_unix_second(-1);

    auto js = timestamp::UNIX_EPOCH_SECONDS -1;
    jd = js / SECS_PER_DAY;

    std::cout << "ep: " << ep._timestamp
              << ", str: " << ep.to_string() 
              << ", jd:" << jd << std::endl;
```

Reply:
thanks, I'll have a try.

Reply:
I've tried following code, but there is still 343 seconds lost.

```cpp

Timestamp timestamp::of_epoch_second(int64_t seconds, int64_t nanoseconds) {
    int64_t second = seconds + timestamp::UNIX_EPOCH_SECONDS;
    JulianDate day = second / SECS_PER_DAY;
    Timestamp s = second % SECS_PER_DAY;
    return timestamp::from_julian_and_time(day, s * USECS_PER_SEC + nanoseconds / NANOSECS_PER_USEC);
}
```